### PR TITLE
Add `promoHasImage` prop documentation to `StoryPromo` `Headline` and `Summary`

### LINKED ISSUE #2604: Add `displayImage` prop documentation to `StoryPromo` `Headline`.
**Is your feature request related to a problem? Please describe.**
`displayImage` on the `StoryPromo` `Headline` has not been documented. Also, `displayImage` defaults to true only for `StoryPromo` but not for `Headline`.

**Describe the solution you'd like**
 Rename `displayImage` prop on the `Headline` to something more meaningful. Looking at 
```jsx
<Headline displayImage> This is the headline </Headline>
```
 It's not clear what `displayImage` does. Does it display an `Image` on the `Headline`

Update `displayImage` prop of `Headline` to be a not required prop that defaults to true.

**Describe alternatives you've considered**
Add `displayImage` prop to documentation.

**Testing notes**
[Tester to complete]

Dev insight: Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by AlistairGempf**
Resolves #2604 

**Overall change:** Add docs to `StoryPromo` `Headline` and `Summary` for the `promohasImage` prop which has been renamed from `displayImage`.

**Code changes:**

- Rename `displayImage` to `promoHasImage` for both the `Headline` and `Summary`
- Add documentation about the `promoHasImage` prop for both the `Headline` and `Summary`
- Update the unit tests to check the no image stories

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: packages/components/psammead-story-promo/src/index.stories.jsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ const InfoComponent = ({
       script={script}
       topStory={topStory}
       service={service}
-      displayImage={displayImage}
+      promoHasImage={displayImage}
```
Comment (DenisHdz) [inline packages/components/psammead-story-promo/src/index.stories.jsx]: You also need to update it in the `Summary`

**Review by DenisHdz — CHANGES_REQUESTED** [review]

**Review by DenisHdz — APPROVED** [review]
LGTM 👍

File: packages/components/psammead-story-promo/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 | ------- | ----------- |
+| 3.0.0-alpha.8 | [PR#2648](https://github.com/bbc/psammead/pull/2648) Rename `displayImage` prop to `promoHasImage` to better indicate it's use |
```
Comment (Bopchy) [inline packages/components/psammead-story-promo/CHANGELOG.md]:
```suggestion
| 3.0.0-alpha.8 | [PR#2648](https://github.com/bbc/psammead/pull/2648) Rename `displayImage` prop to `promoHasImage` to better indicate its use |
```

**Review by j-pendlebury — APPROVED** [review]
LGTM

**Review by Bopchy — APPROVED** [review]

**Pull request merged**